src/datamapper.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2013 Spotify AB
+// Copyright (c) 2018 viper.craft@gmail.com
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+#pragma once
+
+#include <sys/types.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+
+namespace detail {
+  struct DataMapping
+  {
+    void const *data = nullptr;
+    size_t size = 0;
+  };
+
+} // namespace detail
+
+
+class MMapDataMapper {
+public:
+    typedef detail::DataMapping Mapping;
+
+public:
+  Mapping map(const char* filename, bool need_mlock) {
+    int fd = open(filename, O_RDONLY, (int)0400);
+    if (fd == -1) {
+      return Mapping{ nullptr, 0 };
+    }
+    off_t size = lseek(fd, 0, SEEK_END);
+    void *mmaped = mmap(0, size, PROT_READ, MAP_SHARED, fd, 0);
+    close(fd);
+    if (mmaped == MAP_FAILED) {
+      return Mapping{ nullptr, 0 };
+    }
+    if (need_mlock) {
+      mlock(mmaped, size);
+    }
+    return Mapping{ mmaped, (size_t)size };
+  }
+
+  void unmap(const Mapping & mapping) {
+    if (mapping.data) {
+      munmap(const_cast<void*>(mapping.data), mapping.size);
+    }
+  }
+};
+
+
```

Comment:
Need some detailed description here, like:
Anonymous HugeTLB mapping, easy to use, but requires copy from disk(cache) to memory vs HugeTLB FS.